crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -226,10 +242,26 @@ impl EventHandler {
                     hex::encode(payment_hash.0)
                 );
 
-                let mut payments = self.outbound_payments.lock().unwrap();
-                if payments.contains_key(&payment_hash) {
-                    let payment = payments.get_mut(&payment_hash).unwrap();
-                    payment.status = HTLCStatus::Failed;
+                let mut payments = self
+                    .outbound_payments
+                    .lock()
+                    .map_err(|e| anyhow!("{e:#}"))?;
+
+                match payments.entry(payment_hash) {
+                    Entry::Occupied(mut e) => {
+                        let payment = e.get_mut();
+                        payment.status = HTLCStatus::Failed;
+                    }
+                    Entry::Vacant(e) => {
+                        e.insert(PaymentInfo {
+                            preimage: None,
+                            secret: None,
+                            status: HTLCStatus::Failed,
+                            // todo: decode amount from payment hash?
```

Comment:
I don't think you can decode it from a hash? You'd need to retrieve it from somewhere.

Reply:
I think you are right.. otherwise that hash wouldn't be a good hash 😅